lib/tasks/easydeposit2.rake

```diff
@@ -0,0 +1,135 @@
+    (emails, abstract) = find_by_uid_csv(csv, uid)
+    pub.update(abstract: abstract) unless abstract.blank?
+    authors = emails.map { |e| { email: e } }
+    pub.add_author_emails(authors)
+    pub.recruit_authors!
+    # EmailArticleRecruitJob.perform_later(publication: pub)
+  end
+
+end
+
+# Convert WoS export in csv to SOAP XML format, and create Records with the XML
+# @param csv table
+# The row is expected to be double quotes, like: "ZnCl2 ""Water-in-Salt"" Electrolyte Transforms..."
+def create_records(logger, csv)
+  builder = Nokogiri::XML::Builder.new do |xml|
+    xml.return {
+      csv.each do |row|
+        xml.records {
+          xml.uid row[:uid]
+          xml.title {
+            xml.label 'Title'
+            xml.value row[:titles]
+          } unless row[:titles].nil?
+          xml.doctype {
+            xml.label 'Doctype'
+            xml.value row[:doctypes]
+          }
+          xml.source {
+            xml.label 'Issue'
+            xml.value row[:issues]
+          }
+          xml.source {
+            xml.label 'Pages'
+            xml.value row[:pages]
+          }
+          xml.source {
+            xml.label 'Published.BiblioDate'
+            xml.value row[:Biblio_dates]
+          }
+          xml.source {
+            xml.label 'Published.BiblioYear'
+            xml.value row[:Biblio_years]
+          }
+          xml.source {
+            xml.label 'SourceTitle'
+            xml.value row[:source_titles]
+          } unless row[:source_titles].nil?
+          xml.source {
+            xml.label 'Volume'
+            xml.value row[:volumes]
+          }
+          xml.authors {
+            xml.label 'Authors'
+            Array.wrap(row[:authors]).each do |name|
+              xml.value name
+            end
+          } unless row[:authors].nil?
+          xml.keywords {
+            xml.label 'Keywords'
+            Array.wrap(row[:keywords]).each do |keyword|
+              xml.value keyword
+            end
+          } unless row[:keywords].nil?
+          xml.other {
+            xml.label 'Identifier.Doi'
+            xml.value row[:dois]
+          } unless row[:dois].nil?
+          xml.other {
+            xml.label 'Identifier.Issn'
+            xml.value row[:issns]
+          } unless row[:issns].nil?
+        }
+      end
+    }
+  end
+  # logger.info(builder.to_xml)
+  # Need to call Record.new explicitly as inputs for ProcessRecords
+  WebOfScience::Records.new(records: builder.to_xml)
+end
+
+# find emails and abstract from csv for a publication by uid
+# @param csv [CSV table] uid [String]
+# return emails [Array] abstract [String]
+def find_by_uid_csv(csv, uid)
+  emails = []
+  csv.each do |row|
+    if row[:uid].casecmp(uid).zero?
+      if row[:author_emails].include? ";"
+        emails = row[:author_emails].split(';').flatten
+      else
+        emails << row[:author_emails]
+      end
+      abstract = row[:abstract] unless row[:abstract].blank?
```

Comment:
Since you should only have one match in the csv to a uid, it would speed things up to put a return in here.
`return emails, abstract`